Sofa/Component/Constraint/Lagrangian/Model/src/sofa/component/constraint/lagrangian/model/FixedLagrangianConstraint.cpp

```diff
@@ -0,0 +1,143 @@
+/******************************************************************************
+*                 SOFA, Simulation Open-Framework Architecture                *
+*                    (c) 2006 INRIA, USTL, UJF, CNRS, MGH                     *
+*                                                                             *
+* This program is free software; you can redistribute it and/or modify it     *
+* under the terms of the GNU Lesser General Public License as published by    *
+* the Free Software Foundation; either version 2.1 of the License, or (at     *
+* your option) any later version.                                             *
+*                                                                             *
+* This program is distributed in the hope that it will be useful, but WITHOUT *
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#include <sofa/component/constraint/lagrangian/model/FixedLagrangianConstraint.inl>
+
+#include <sofa/defaulttype/VecTypes.h>
+#include <sofa/core/ObjectFactory.h>
+#include <sofa/component/constraint/lagrangian/model/BilateralConstraintResolution.h>
+
+namespace sofa::component::constraint::lagrangian::model
+{
+
+using namespace sofa::defaulttype;
+using namespace sofa::helper;
+
+
+// Vec3D specialization
+template<>
+void FixedLagrangianConstraint<Vec3Types>::doBuildConstraintLine( MatrixDeriv &c, unsigned int &cIndex, unsigned int lineNumber)
+{
+    constexpr Coord cx(1,0,0), cy(0,1,0), cz(0,0,1);
+    unsigned dofIdx = d_fixAll.getValue() ? lineNumber : d_indices.getValue()[lineNumber];
+
+    cIndex += 3;
```

Comment:
I would prefer `Deriv::total_size` instead of 3. Also valid for rigid types.

Reply:
Yeap, I wanted to use such generic attribute, but I struggled to find it, so I was happy with the specialization. Thank's, I can then put it out of the specialization.